src/pages/DataSubmissionForm.js

```diff
@@ -41,50 +41,88 @@ export const DataSubmissionForm = () => {
     init();
   }, []);
 
-  let formData = {};
+  const formData = {};
+
+  // separate json object and files to send to consent
+  const separateRegistrationAndFiles = (formData) => {
+    const registration = cloneDeep(formData);
+    const files = {};
+
+    if (!isEmpty(registration.alternativeDataSharingPlanFile)) {
+      files.alternativeDataSharingPlan = registration.alternativeDataSharingPlanFile;
+      delete registration.alternativeDataSharingPlanFile;
+    }
+
+    const consentGroups = registration['consentGroups'];
+
+    for (let i = 0; i < consentGroups?.length; i++) {
```

Comment:
nit: prefer map with index over for loops.

Reply:
I had this thought too. Too bad Java doesn't have a built-in in `Enumerate` method like Python!